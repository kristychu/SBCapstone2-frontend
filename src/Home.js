import React from 'react';
import { Link } from "react-router-dom";
import { useSelector } from 'react-redux';
import { Button } from 'reactstrap';

function SkincareAppHome(){
    const token = useSelector(st => st.users.profile.token)
    if(token){
        const decoded = JSON.parse(atob(token.split('.')[1]));
        const username = decoded.username
        return(
            <div>
                <h2>Welcome back to the Korean Skincare App, {username}! </h2>
                <p>We're excited for you to join us!</p>
                <Link to="/user-routine"><Button color="success">Get Started</Button></Link>
            </div>
        )
    }
    return(
        <div>
            <h2>Welcome to the Korean Skincare App</h2>
            <p>Track your skincare products for each step of the <Link to="/korean-skincare-routine">Korean Skincare Routine!</Link></p>
            <p>Have an account already? <Link to="/login">Login!</Link></p>
            <p>Otherwise, <Link to="/register">Register here</Link>.</p>
        </div>
    )
}

export default SkincareAppHome;