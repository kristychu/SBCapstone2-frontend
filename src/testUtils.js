import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

export const makeMockStore = configureStore([thunk]);
