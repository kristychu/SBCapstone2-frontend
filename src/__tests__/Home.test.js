import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import Home from "../Home";
import { MemoryRouter } from "react-router";
import { makeMockStore } from "../testUtils";

//USING ACTUAL REDUCER WHICH SHOULD BE EMPTY
const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot for anonymous user", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

// CREATING A MOCK STORE AND FAKED DATA
const mockStore = makeMockStore({
  users: {
    profile: {
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwiaWF0IjoxNjE4NDQzNDQ0fQ.8PZOXva_Bztp48k8obLY5sj8k7siJVMTIhftj6509EI",
    },
  },
});

it("should match snapshot if logged in", function () {
  const { asFragment } = render(
    <Provider store={mockStore}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
