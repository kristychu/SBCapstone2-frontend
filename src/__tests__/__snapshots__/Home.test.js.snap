// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for anonymous user 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Welcome to the Korean Skincare App
    </h2>
    <p>
      Track your skincare products for each step of the 
      <a
        href="/korean-skincare-routine"
      >
        Korean Skincare Routine!
      </a>
    </p>
    <p>
      Have an account already? 
      <a
        href="/login"
      >
        Login!
      </a>
    </p>
    <p>
      Otherwise, 
      <a
        href="/register"
      >
        Register here
      </a>
      .
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot if logged in 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Welcome back to the Korean Skincare App, testuser! 
    </h2>
    <p>
      We're excited for you to join us!
    </p>
    <a
      href="/user-routine"
    >
      <button
        class="btn btn-success"
      >
        Get Started
      </button>
    </a>
  </div>
</DocumentFragment>
`;
